validator/sawtooth_validator/journal/responder.py

```diff
@@ -44,6 +53,31 @@ def check_for_batch_by_transaction(self, transaction_id):
         batch = self.completer.get_batch_by_transaction(transaction_id)
         return batch
 
+    def add_request(self, requested_id, connection_id):
+        with self._lock:
+            if (requested_id in self.pending_requests) and \
+                    (connection_id not in self.pending_requests[requested_id]):
+                self.pending_requests[requested_id] += [connection_id]
+
+            else:
+                self.pending_requests[requested_id] = [connection_id]
```

Comment:
Good catch, that if should be separated into two, where if the connection id is already in the list, nothin should happen. Will update.